openedx/features/qverse_features/registration/signals.py

```diff
@@ -98,22 +99,30 @@ def create_users_from_csv_file(sender, instance, created, **kwargs):
                          .format(instance.admission_file.path, err))
     csv_file.close()
     _write_status_on_csv_file(instance.admission_file.path, output_file_rows)
-    new_students = [student for student in output_file_rows if student.get('status') == USER_CREATED]
+    new_students = [student for student in output_file_rows
+                    if student.get('status') == USER_CREATED or
+                    (student.get('status') == USER_UPDATED and student.get('regno') in users_with_updated_emails)]
     site = get_current_site()
     protocol = 'https' if get_current_request().is_secure() else 'http'
     send_bulk_mail_to_newly_created_students.delay(new_students, site.id, protocol)
 
 
-def _create_or_update_edx_user(user_info):
+def _create_or_update_edx_user(user_info, users_with_updated_emails):
     """
     Creates/Updates edx user from given information.
 
     Arguments:
         user_info (dict): A dict containing user information
+        users_with_updated_emails (set): A set containing registration numbers of students whose emails addresses
+                                         have been updated
 
     Returns:
         edx_user (User): A newly created/updated User object
     """
+    is_new_email_address = False
+    if not User.objects.filter(email=user_info.get('email')).exists():
```

Comment:
What if we get the users matching the provided list at once and then do the filtering to avoid multiple DB hits ? just a thought